File: csharp/Workers/CreditDeductionWorker.cs
using Camunda.Orchestration.Sdk;
using Camunda.Training.CSharp.Services;

namespace Camunda.Training.CSharp.Workers
{
    // Input/output DTOs. Property names map to the BPMN process variables
    // (System.Text.Json matches them case-insensitively, so CustomerId <-> customerId).
    public record DeductionInput(string CustomerId, double OrderTotal);
    public record DeductionOutput(double OpenAmount1);

    public class CreditDeductionWorker(CamundaClient client) : Worker("credit-deduction", client)
    {
        public override Task<object?> Handler(ActivatedJob job, CancellationToken ct)
        {
            Console.WriteLine($"Handling credit-deduction job: {job.JobKey}");

            // Read the process variables into a typed DTO.
            var input = job.GetVariables<DeductionInput>();
            Console.WriteLine($"Variables: {input}");

            CustomerService customerService = new CustomerService();
            double customerCredit = customerService.GetCustomerCredit(input!.CustomerId);
            double openAmount = customerService.DeductCredit(customerCredit, input.OrderTotal);

            // Returning this DTO auto-completes the job with openAmount.
            return Task.FromResult<object?>(new DeductionOutput(openAmount));
        }
    }
}

File: java/src/main/java/io/camunda/training/workers/CreditDeductionWorker.java
package io.camunda.training.workers;

import io.camunda.client.api.response.ActivatedJob;
import io.camunda.client.api.worker.JobClient;
import io.camunda.client.api.worker.JobHandler;
import io.camunda.training.services.CustomerService;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.Map;

public class CreditDeductionWorker implements JobHandler {

    private static final Logger logger = LoggerFactory.getLogger(CreditDeductionWorker.class);

    private final CustomerService customerService;

    public CreditDeductionWorker(CustomerService customerService) {
        this.customerService = customerService;
    }

    @Override
    public void handle(JobClient client, ActivatedJob job) throws Exception {
        logger.info("Job handled: {}", job.getType());

        Map<String, Object> variables = job.getVariablesAsMap();
        String customerId = (String) variables.get("customerId");
        Number orderTotal = (Number) variables.get("orderTotal");

        Double openAmount = customerService.deductCredit(customerId, orderTotal.doubleValue());

        client.newCompleteCommand(job).variables(Map.of("openAmount", openAmount)).send();
    }
}
